core_gems.rb

```diff
@@ -1,13 +1,16 @@
 dir 'core_gems'
 download "json", "2.1.0"
-download "msgpack", "1.2.2"
-download "cool.io", "1.5.3"
+download "msgpack", "1.2.9"
+download "cool.io", "1.5.4"
 download "http_parser.rb", "0.6.0"
 download "yajl-ruby", "1.3.1"
 download "sigdump", "0.2.4"
 download "thread_safe", "0.3.5"
 download "oj", "3.3.10"
 download "tzinfo", "1.2.2"
 download "tzinfo-data", "1.2016.5"
-fetch "google-protobuf", "3.7.1"
-fetch "grpc", "1.20.0"
+unless windows?
+  # Build dependencies for 'google-fluentd' that aren't needed on Windows.
+  fetch "google-protobuf", "3.9.0"
```

Comment:
[Optional] It might be helpful to delay gem version bumps unless they are required by this change. Because they might introduce extra noise and potentially break nightly jobs (grpc and protobuf updates had broken us multiple times in the past). This way it's clear if a certain issue is isolated to this refactoring only or some new gem dependencies.